Sources/Plasma/NucleusLib/pnAsyncCore/Private/pnAcThread.h

```diff
@@ -65,10 +66,40 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 // for IoWaitId/TimerCreate/TimerUpdate
 constexpr unsigned kAsyncTimeInfinite = (unsigned) -1;
 
+struct AsyncThread;
+
+struct AsyncThreadRef {
+    std::shared_ptr<AsyncThread> impl;
```

Comment:
I think that change can happen. It couldn't in earlier iterations that are irrelevant now. std::thread will invalidate any old value once moved, so the shared pointer was to insulate against that issue when the value gets stored, or gets provided to a thread. The AsyncThreadRef structure provides that insulation now in since impl is a shared_ptr.